packages/eas-cli/src/commands/build/run.ts

```diff
@@ -143,6 +144,11 @@ async function maybeGetBuildAsync(
   projectId: string,
   paginatedQueryOptions: PaginatedQueryOptions
 ): Promise<BuildFragment | null> {
+  const distributionType =
+    flags.selectedPlatform === AppPlatform.Ios
+      ? DistributionType.Simulator
+      : DistributionType.Internal;
```

Comment:
why? you can install any android build in emulator, not only internal distribution

Reply:
@brentvatne How do you think this filtering should work? We had a conversation about it today with @wkozyra95  and it looks like currently there is no trivial way to retrieve only builds containing `.apk` from the backend. 

I see some potential options:

1. **Add server-side filtering**: We can create an index on the `artifacts` column in the database (which is `JSONB` type) and filter builds on this column using regex to see if the `applicationArchiveUrl` contains `.apk`, which doesn't sound like a perfect solution, to be honest. It feels like it would cause some performance issues.
2. **Add client-side filtering**: We can query for all Android builds and filter them on the `build.artifacts.applicationArchiveUrl` field as above. In some cases, we would need to hit the backend multiple times so the number of builds left after filtering is equal to the `limit` parameter. The factor limiting the number of queries we make to the backend, in this case, is the fact that the archives for builds older than 30 days get deleted, so it should be a reasonable number. It's also not a perfect solution.
3. **Use some easier, but more inaccurate filtering**: We can use simple backend filters like `distributionType === INTERNAL`, but it can result in query results containing `.aab` builds and missing on some `.apk` builds as well. It feels like it would work correctly for the majority of the builds, although it's still an inaccurate solution and it might be confusing for the users.

What do you think about it?

CC @dsokal

Reply:
Ad 1. I'm reworking that part currently in www that value can URL/bucket key. We would need to have an index on a column with a json object, which is not great.
Ad. 2. Client-side filtering is a bad idea in most cases, but here it could work. Normally the problem is that if you want to show 10 builds you would need to query in a loop until you get those builds or check all builds. In most cases, it would not be acceptable to abuse API like that, but here we only need builds that artifacts did not expire, so upper limit on number of requests is not that high.
Ad. 3 It's maybe not 100% correct solution, but I think skipping filtering by artifact type is the best approach here. I'm not sure whether we should filter only internal distribution builds or just take all android builds. With this approach, we would just list the latest builds(or the latest internal distribution) without any validation, and if the user selects a build that produced aab, we would just throw an error then.

Reply:
I like the synchronicity of this discussion! I had this on my tasks for the day:

> Create task: when aab is built for internal distribution, don't let people download and try to install

We had a developer email us with some confusion around installing an internal distribution build on their device. It turned out that they had configured the build to produce an aab rather than the default apk for internal distribution, but on our build page we didn't provide any indication that this may be a problem, and we provided the usual UI for them to download and attempt to launch it on their device. It turned out that they had an app installed that was registered to launch files with the aab extension, so when they opened the downloaded aab it opened another application entirely, creating grave concern.

<hr />

Here's another possible idea - what if we showed all builds in the list but made any aab's unselectable and made that visually clear? This would be workable as a first pass and it would also allow us to determine whether this is annoying for users. It also helps to communicate that aab's are not directly installable to devices and apks are. If we don't show aabs in the list, I could imagine some users running this command and being confused as to why they don't see a particular build. This allows us to communicate that directly, at the cost of having to possibly sort through more builds in the list. Ultimately we will add some filters to this in the future, such as filter by development client builds, and so users can take care of filtering out aabs on their own.

Reply:
> Here's another possible idea - what if we showed all builds in the list but made any aab's unselectable and made that visually clear

I like the idea. My only concern is what if all builds are unselectable? 🤔

Reply:
> My only concern is what if all builds are unselectable?

We are already doing that for submits, when artifacts expire we still show those builds but it's crossed out.

Reply:
I also like this idea, although I wonder if this is a problem if the command behavior is inconsistent between iOS and Android. Meaning that on Android we show all builds with some of them in an unselectable state, but on iOS, we show only filtered simulator builds. I'm wondering if this is acceptable in your opinion 🤔 (I assume that it is, although I want to double-check).

Reply:
>We are already doing that for submits, when artifacts expire we still show those builds but it's crossed out.

@wkozyra95 I think we log an error in such a case.

Here is the code which handles it: https://github.com/expo/eas-cli/blob/main/packages/eas-cli/src/submit/ArchiveSource.ts#L260

<img width="510" alt="Screenshot 2022-11-03 at 11 17 19" src="https://user-images.githubusercontent.com/55145344/199696000-a79d6d99-8c04-4b8e-8f74-2094d4be8c74.png">

Reply:
This is how it looks now:

<img width="773" alt="Screenshot 2022-11-04 at 09 45 39" src="https://user-images.githubusercontent.com/55145344/199930983-d7e55f55-9518-4066-9f55-5a5f01afa026.png">
<img width="774" alt="Screenshot 2022-11-04 at 09 45 51" src="https://user-images.githubusercontent.com/55145344/199931015-397f3f9e-0cca-4939-9994-c96eb6e750a3.png">


Please notice, you don't see the underline issues on these screen shots because I'm using VSCode. The problem still exists for other terminals.
https://github.com/expo/eas-cli/pull/1502#issuecomment-1302127360